### ISSUE TITLE
Laravel 5.6 : Unknown custom element: <media-manager>

### ISSUE THREAD

**Issue by partikule** [talvbansal/media-manager#81]
The component <media-manager> is not recognized when using in a sub component of an application.

Here are the simple steps to reproduce the problem.

I guess the problem comes from the /vendor/talvbansal/media-manager/resources/assets/js/base.js file.

We tried to change our app Vue load, without success.

**resources/assets/js/app.js :** 

```
import router from './router';

require('./bootstrap');

import Vue from 'vue';

// Add this line...
require('./../talvbansal/media-manager/js/media-manager');

import App from './App.vue';


new Vue({
    el: '#app',
    router: router,
    render: h => h(App)
});

```

**resources/assets/js/App.vue :**

```
<template>
    <div>
        <router-view/>
    </div>
</template>

<script type="text/javascript">

    export default {
        name: 'app'
    }

</script>
```

**resources/assets/js/router.js :** 

```

import Vue from 'vue';
import Router from 'vue-router';

import Media from './components/MediaComponent.vue';

Vue.use(Router);

const router = new Router({
    mode: 'history',
	base: '/component',
    routes: [
        {
            path: '/media',
            name: 'mediaComponent',
            component: Media
        }
    ]
});

export default router;

```

**resources/assets/js/components/MediaComponent.vue** 

```
<template>
    <div>
        <h2>Here we should have the media manager :</h2>
		<media-manager></media-manager>
    </div>
</template>

<script>

	export default {
        name: 'mediaComponent'
    }
</script>

```

**routes/web.php**

```
<?php

/*
|--------------------------------------------------------------------------
| Web Routes
|--------------------------------------------------------------------------
|
| Here is where you can register web routes for your application. These
| routes are loaded by the RouteServiceProvider within a group which
| contains the "web" middleware group. Now create something great!
|
*/

Route::get('/', function () {
    return view('welcome');
});


Route::get('/component/{sublevels?}', function(){
    return view('media-manager');
})->where('sublevels', '.*');

\TalvBansal\MediaManager\Routes\MediaRoutes::get();
```

**resources/views/media-manager.blade.php**

```
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">

    <title>Media Manager Test</title>

    <meta name="csrf-token" content="{{ csrf_token() }}">

    <link rel="stylesheet" href="/css/app.css" rel="stylesheet" type="text/css">

</head>
<body>

<h1>Media Manager Demo</h1>

<div id="app"></div>

<script src="/js/app.js"></script>

</body>
</html>
```

**Comment by partikule** [comment]
For the moment, the only way we found to use it in our app is to copy the source in our own components folder and to register each component through an install.js file.

**install.js :** 

```
import Vue from 'vue';

import FileManagerMixin from './mixins/file-manager-mixin';

import MediaModal from './components/MediaModal.vue';
import MediaCreateFolderModal from './components/CreateFolderModal.vue';
import MediaDeleteItemModal from './components/ConfirmDeleteModal.vue';
import MediaErrors from './components/Errors.vue';
import MediaMoveItemModal from './components/MoveItemModal.vue';
import MediaRenameItemModal from './components/RenameItemModal.vue';
import MediaManager from './components/MediaManager.vue';

const install = () => {

	Vue.mixin(FileManagerMixin);

	Vue.component('MediaModal', MediaModal);
	Vue.component('MediaCreateFolderModal', MediaCreateFolderModal);
	Vue.component('MediaDeleteItemModal', MediaDeleteItemModal);
	Vue.component('MediaErrors', MediaErrors);
	Vue.component('MediaMoveItemModal', MediaMoveItemModal);
	Vue.component('MediaRenameItemModal', MediaRenameItemModal);
	Vue.component('MediaManager', MediaManager);
};

install();

export default { install };

```

We then use it like this in the target component : 

```
<template>
    <div>
        <h2>Here we should have the media manager :</h2>
		<media-manager></media-manager>
    </div>
</template>

<script>

	import MediaManager from './media-manager/';

	export default {
        name: 'mediaComponent',
		components: {
		}
    }
</script>

```

Btw, several problems remains : 
- The registration of window.eventHub (on the window object) can be unnecessary overload if the main app already has one Event Bus and if this Event Bus is named differently than "eventHub".
- jQuery is used to get the dropzone element. This make the component unusable for apps who don't use jQuery but Vue.js with vanilla JS.

**Comment by talvbansal** [comment]
Hi @partikule, 

Thanks for reporting your finds i'll have a look into them. 

If you know how to fix any of the issues you've outlined I'd really welcome pull requests :)

Thanks

Talv

**Comment by partikule** [comment]
Hi @talvbansal,

Thanks for this positive answer !

We're currently including your media-manager in our "main big backend app".
I will see if we have the time to fork / modify and make pull requests.
For the moment, we're arranging it a little, as we use Bulma and not bootstrap.

Btw, if you're interested in, I can open issues and give you our little enhancements when we found some to implement.

**Comment by talvbansal** [comment]
That would be amazing @partikule  - im looking into some of the above now :)

**Comment by talvbansal** [comment]
Fixed the jquery dependency with 46d8c52

**Comment by talvbansal** [comment]
@partikule i really want to find a way to make this project framework agnostic - have you any ideas or are you just replacing classes within your instance to be the appropriate bulma classes?
